tools/schemacode/schemacode/validator.py

```diff
@@ -658,6 +675,77 @@ def log_errors(validation_result):
         lgr.warning("The `%s` file was not matched by any regex schema entry.", i)
 
 
+def _inheritance_expansion(
+    regex_string,
+    datatype=None,
+    debug=False,
+):
+    """
+    Generate regex strings applying BIDS inheritance expansion to an input string.
+
+    Parameters
+    ----------
+    regex_string : str
+        String representing the regex to which inheritance expansion should be applied.
+    datatype : str, optional
+        Datatype string to remove as part of inheritance expansion.
+    debug : bool, optional
```

Comment:
let's not introduce `debug` options to functions.  There is `lgr.debug` which could be "unconditionally" used to log desired debug information, and that is already used in this module

```shell
$> git grep '\.debug' -- tools/schemacode/schemacode/
tools/schemacode/schemacode/schema.py:        lgr.debug(f"Loading {object_group_file.stem} objects.")
tools/schemacode/schemacode/schema.py:        lgr.debug(f"Loading {rule_group_file.stem} rules.")
tools/schemacode/schemacode/schema.py:        lgr.debug(f"Loading {rule_group_file.stem} rules.")
tools/schemacode/schemacode/tests/conftest.py:        lgr.debug("Cloning %r into %r", url, path)
```

there is also env var which can adjust (not sure why we have two invocations since render probably imports schema anyways)

```
$> git grep set_logger_level
tools/schemacode/schemacode/render.py:utils.set_logger_level(lgr, os.environ.get("BIDS_SCHEMA_LOG_LEVEL", logging.INFO))
tools/schemacode/schemacode/schema.py:utils.set_logger_level(lgr, os.environ.get("BIDS_SCHEMA_LOG_LEVEL", logging.INFO))
tools/schemacode/schemacode/utils.py:def set_logger_level(lgr, level):
```

Reply:
Might have been better as a separate PR since it's not specific to inheritance support... but given that PRs are a bit slow, might as well add it in here :)